hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/state/components/NetworkInfoImpl.java

```
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.web3.state.components;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.mirror.web3.evm.properties.MirrorNodeEvmProperties;
+import com.hedera.node.app.info.SelfNodeInfoImpl;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.state.spi.info.NetworkInfo;
+import com.swirlds.state.spi.info.NodeInfo;
+import com.swirlds.state.spi.info.SelfNodeInfo;
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import jakarta.inject.Named;
+import java.util.Collections;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+
+@Named
+@RequiredArgsConstructor
+public class NetworkInfoImpl implements NetworkInfo {
+
+    private final MirrorNodeEvmProperties mirrorNodeEvmProperties;
+
+    @Nonnull
+    @Override
+    public Bytes ledgerId() {
```

Are we sure that ledgerId and addressBook() aren't used by EVM calls? We have the ledgerId in the properties we could return it.

If they're not used, maybe we should throw exceptions instead of return empty objects so we can catch problems earlier in testing?

---

We can't really be sure if they will be used or not until we start testing executing transaction.
Throwing exceptions for now for these methods seems pretty reasonable so I applied it. 

And once we start testing if we hit those exceptions then we'll implement them.